File: second-year/second-semester/MPP/mpp-proiect-repository-raulbaciulescu/travelAgency2/travelAgency2/Repository/UserRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using travelAgency2.Repository.Database;
using travelAgency2.src.Domain;
using travelAgency2.src.Repository;
using travelAgency2.Utils;

namespace travelAgency2.Repository
{
    internal class UserRepository : Repository<long, User>
    {
        private UserTable table;

        public UserRepository()
        {
            table = (UserTable) Resources.getTableFactory().getTable(Constants.Db.Tables.USER);
         }
           
        public void Add(User entity)
        {
            table.Add(entity);
        }

        public void Delete(long id)
        {
            throw new NotImplementedException();
        }

        public User FindByID(long id)
        {
            return table.FindById(id);
        }

        public List<User> GetAll()
        {
            return table.GetAll();
        }

        public void Update(User entity, User entityNew)
        {
            throw new NotImplementedException();
        }

        internal User findUser(string username, string password)
        {
            return table.findUser(username, password);
        }
    }
}

File: second-year/first-semester/MAP/SocialApp/src/main/java/com/example/socialapp/repository/UserRepository.java
package com.example.socialapp.repository;

import com.example.socialapp.database.TableFactory;
import com.example.socialapp.domain.User;

import com.example.socialapp.database.table.UserTable;
import com.example.socialapp.utils.Constants;

import java.sql.SQLException;
import java.util.ArrayList;
import java.util.Optional;

public class UserRepository implements Repository<Long, User> {
    private final UserTable table;

    public UserRepository() throws SQLException {
        this.table = (UserTable) TableFactory.getInstance().table(Constants.Tables.USERS);
    }

    @Override
    public void add(User user) throws SQLException {
        table.insert(user);
    }

    @Override
    public void update(User user) throws SQLException {
        table.update(user);
    }

    @Override
    public Optional<User> findByID(Long id) throws SQLException {
        return table.findByID(id);
    }

    @Override
    public void delete(Long id) throws SQLException {
        table.delete(id);
    }

    @Override
    public ArrayList<User> getAll() throws SQLException {
        return table.getAll();
    }
}
